server/conn.go

```diff
@@ -1335,13 +1369,8 @@ func (cc *clientConn) handleFieldList(sql string) (err error) {
 // If binary is true, the data would be encoded in BINARY format.
 // serverStatus, a flag bit represents server information.
 // fetchSize, the desired number of rows to be fetched each time when client uses cursor.
-// resultsets, it's used to support the MULTI_RESULTS capability in mysql protocol.
 func (cc *clientConn) writeResultset(ctx context.Context, rs ResultSet, binary bool, serverStatus uint16, fetchSize int) (runErr error) {
 	defer func() {
-		// close ResultSet when cursor doesn't exist
-		if !mysql.HasCursorExistsFlag(serverStatus) {
-			terror.Call(rs.Close)
-		}
```

Comment:
Should this change be involved with `clientConn.handleStmtExecute`?

Reply:
The normal logic never run here.
This is a special protocol used by CMD_STMT_FETCH, it makes the code ugly.
I've recover the old code, and @lysu would add an integration test for it in the future (this can't be done in golang because golang mysql lib doesn't support this protocol).

Reply:
https://github.com/pingcap/tidb-test/pull/1037